uvtools/dspec.py

```diff
@@ -1661,29 +1685,40 @@ def _clean_filter(x, data, wgts, filter_centers, filter_half_widths,
     info['filter_params'] = {'axis_0':{}, 'axis_1':{}}
     info['clean_status'] = {'axis_0':{}, 'axis_1':{}}
     info['status'] = {'axis_0':{}, 'axis_1':{}}
-    if filt2d_mode == 'rect' or not clean2d:
+    if not clean2d:
+        # if we are not in 2d clean mode, then area is the outer product of the
+        # cleaning area in the axis to be cleaned and ones along the axis not to
+        # be cleaned. This is because the filter_width is infinity along the axis
+        # not to be cleaned so _get_filter_area gives all ones.
         for m in range(2):
             for fc, fw in zip(filter_centers[m], filter_half_widths[m]):
                 area_vecs[m] = _get_filter_area(x[m], fc, fw)
         area = np.outer(area_vecs[0], area_vecs[1])
-    elif filt2d_mode == 'plus' and clean2d:
+    else:
+        # if we are doing 2dclean, then either do rectangular or plus mode.
         area = np.zeros(data.shape)
-        #construct and add a 'plus' for each filtering window pair in each dimension.
-        for fc0, fw0 in zip(filter_centers[0], filter_half_widths[0]):
-            for fc1, fw1 in zip(filter_centers[1], filter_half_widths[1]):
-                area_temp = np.zeros(area.shape)
-                if fc0 >= _x[0].min() and fc0 <= _x[0].max():
+        # in rectangle mode, the list of time filter centers / widths and frequency filter centers / widths
+        # specify rectangular regions in 2d Fourier space to CLEAN in. This loop iterates over those centers
+        # and widths.
+        for fc_t, fw_t, fc_f, fw_f in zip(filter_centers[0], filter_half_widths[0], filter_centers[1], filter_half_widths[1]):
+            if filt2d_mode == 'rect':
+                # determine rectangular region by taking outer products of filter areas along fourier time and fourier freq
+                # axes.
+                area_t = np.outer(_get_filter_area(x[0], fc_t, fw_t), _get_filter_area(x[1], fc_f, fw_f))
+            elif filt2d_mode == 'plus':
```

Comment:
Inconsistency here with what's written in the docstring: docstring says that supported modes are `'rect'` and `'outer'`, but this uses `'rect'` and `'plus'`.